src/industry_cmd.cpp

```diff
@@ -1126,9 +1126,18 @@ static void ProduceIndustryGoods(Industry *i)
 		uint32 r;
 		uint num;
 		if (Chance16R(1, 14, r) && (num = indsp->number_of_sounds) != 0 && _settings_client.sound.ambient) {
-			SndPlayTileFx(
-				(SoundFx)(indsp->random_sounds[((r >> 16) * num) >> 16]),
-				i->location.tile);
+			bool playsound = false;
+			for (size_t j = 0; j < lengthof(i->last_month_production); j++) {
+				if (i->last_month_production[j] > 0) {
+					playsound = true;
+					break;
+				}
+			}
+
+			if (playsound)
+				SndPlayTileFx(
+					(SoundFx)(indsp->random_sounds[((r >> 16) * num) >> 16]),
+					i->location.tile);
```

Comment:
Missing { } for the if-block, but the SndPlayTileFx() call could just be before the break; above.

Reply:
Your observation is correct - thank you! As for the lack of braces, they aren't required if the `if` statement only has one line inside. I don't think that is a style requirement for OpenTTD since I have seen it being done elsewhere in the code.

Reply:
According to our codestyle, the braces are required whenever the if spans more than one line.